pageserver/src/layered_repository.rs

```diff
@@ -1339,6 +1339,10 @@ impl LayeredTimeline {
         //
         let mut layers = self.layers.lock().unwrap();
         'outer: for l in layers.iter_historic_layers() {
+            if l.is_open() {
```

Comment:
let's add a small comment here, explaining what would go wrong if you removed an open layer.